server/src/main/scala/tailcall/server/GenericServer.scala

```diff
@@ -14,13 +14,20 @@ object GenericServer {
     Http.collectZIO[Request] { case req @ Method.POST -> !! / "graphql" / id =>
       for {
         schema <- if (id == "0") SchemaRegistry.list(0, 1).map(_.headOption) else SchemaRegistry.get(Digest.fromHex(id))
-        result <- schema match {
+        timeout     <- ZIO.succeed(
+          (for {
+            blueprint <- schema
+            timeout   <- blueprint.server.globalResponseTimeout
+          } yield timeout).getOrElse(10000)
+        )
```

Comment:
Can further delay this check, until it's confirmed that the blueprint was found.